clusterous/cluster.py

```diff
@@ -987,55 +1202,122 @@ def instances_terminated():
         return True
 
     def terminate_cluster(self, leave_shared_volume, force_delete_shared_volume):
+        # Connect to EC2 services
         conn = boto.ec2.connect_to_region(self._config['region'],
                     aws_access_key_id=self._config['access_key_id'],
                     aws_secret_access_key=self._config['secret_access_key'])
 
-        if not conn:
+        # Connect to VPC services
+        vpc_conn = boto.vpc.connect_to_region(self._config['region'],
+                    aws_access_key_id=self._config['access_key_id'],
+                    aws_secret_access_key=self._config['secret_access_key'])
+
+        if not (conn or vpc_conn):
             raise ClusterException('Cannot connect to AWS')
 
+        # Delete instances
         instance_list = self._get_instances(self.cluster_name, connection=conn)
         num_instances = len(instance_list)
         instances = [ i.id for i in instance_list ]
+        if instances:
+            self._logger.info('Terminating {0} instances'.format(num_instances))
+            self._terminate_instances_and_wait(conn, instances)
 
-        if not instances:
-            self._logger.info('Nothing to terminate. Cleaning up.')
-            self._delete_cluster_info()
-            return True
```

Comment:
As mentioned, this should be undeleted. If there are no running instances, this method should just clean up the local files and return.